lib/types.js

```diff
@@ -678,9 +685,36 @@ Dp.finalize = function() {
 
         // A linearization of the inheritance hierarchy.
         populateSupertypeList(this.typeName, this.supertypeList);
+
+        var exp = Type.def("Expression");
+        // Calling `exp.isSupertypeOf(...)` will throw if `exp` has not been finalized.
+        // We finalize the entire supertype chain above, so if `exp` is not yet finalized
+        // we can safely assume that this is not an Expression.
+        if (this.buildable && exp.finalized && exp.isSupertypeOf(this)) {
+            wrapExpressionBuilderWithStatement(this.typeName);
+        }
     }
 };
 
+// Adds an additional builder for Expression subtypes
+// that wraps the built Expression in an ExpressionStatements.
+function wrapExpressionBuilderWithStatement(typeName) {
+    var wrapperName = getStatementBuilderName(typeName);
+
+    // skip if the builder already exists
+    if (builders[wrapperName]) return;
+
+    // the builder function to wrap with builders.ExpressionStatement
+    var wrapped = builders[getBuilderName(typeName)];
+
+    // skip if there is nothing to wrap
+    if (!wrapped) return;
+
+    builders[wrapperName] = function() {
```

Comment:
Looking at existing code, I am guessing I should have used `Object.defineProperty` here.